trace/zipin_tracereporter_test.go

```diff
@@ -35,12 +36,49 @@ import (
 )
 
 func TestZipkinTraceReporterFactory(t *testing.T) {
-	_, err := tchannel.NewChannel("client", &tchannel.ChannelOptions{
-		Logger:               tchannel.SimpleLogger,
+	ch, err := tchannel.NewChannel("svc", &tchannel.ChannelOptions{
 		TraceReporterFactory: ZipkinTraceReporterFactory,
 	})
-
 	assert.NoError(t, err)
+
+	// Create a TCollector channel, and add it as a peer to ch so Report works.
+	mockServer := new(mocks.TChanTCollector)
+	tcollectorCh, err := setupServer(mockServer)
+	require.NoError(t, err, "setupServer failed")
+	ch.Peers().Add(tcollectorCh.PeerInfo().HostPort)
+
+	called := make(chan int)
+	ret := &gen.Response{Ok: true}
+	mockServer.On("Submit", mock.Anything, mock.Anything).Return(ret, nil).Run(func(args mock.Arguments) {
+		called <- 1
+	})
+
+	// Make some calls, and validate that the trace reporter is not called recursively.
+	require.NoError(t, json.Register(tcollectorCh, json.Handlers{"op": func(ctx json.Context, arg map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
Was the simplest to set up an inline server to listen on (just needed any listener)